swim/memberlist.go

```diff
@@ -79,11 +84,21 @@ func (m *memberlist) ComputeChecksum() {
 	startTime := time.Now()
 	m.members.Lock()
 	checksum := farm.Fingerprint32([]byte(m.GenChecksumString()))
+	oldChecksum := m.members.checksum
 	m.members.checksum = checksum
 	m.members.Unlock()
+
+	if oldChecksum != checksum {
```

Comment:
I think it will actually be good to have this branch tested. If only to make sure the code works and does not dereference things. I will work on testing all log lines.